carpet/metrics.py

```diff
@@ -0,0 +1,65 @@
+""" Utilities to generate a synthetic 1d data. """
+# Authors: Hamza Cherkaoui <hamza.cherkaoui@inria.fr>
+# Authors: Thomas Moreau <thomas.moreau@inria.fr>
+# License: BSD (3-clause)
+
+import warnings
+import torch
+import numpy as np
+import prox_tv
+from .utils import init_vuz
+from .checks import check_tensor
+from .lista_analysis import ListaTV
+
+
+def compute_prox_tv_errors(network, x, lbda):
+    """ Returnt the sub-optimality gap of the prox-tv at each iteration. """
+
+    if not isinstance(network, (ListaTV)):
+        raise ValueError("network should be {'ListaTV',}")
+
+    if not hasattr(network, 'training_loss_'):
+        warnings.warn("network seems to not have been trained "
+                      "training_loss_ attribute missing")
+
+    def tv_loss(x, u, lbda):
+        """ TV reg. loss function for Numpy variables. """
+        n_samples = u.shape[0]
+        data_term = 0.5 * np.sum(np.square(u - x))
+        reg = lbda * np.sum(np.abs(np.cumsum(u)))
```

Comment:
I though `u` is the primal variable for the analysis so this should be a diff no?

```suggestion
        reg = lbda * np.sum(np.abs(np.diff(u)))
```